### PR TITLE
[SYCL] [Joint Matrix] Modularized tests

### PR DESCRIPTION
**Pull request by artemrad**
This is part 1 of 2 to improve reuse of common functions in the Joint Matrix test set. This changes focuses on removing functions, aliases and types from individual tests and relying on them being present in `common.hpp`.

### CODE REVIEW

File: sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-#include <cstddef>
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp:14]: cstddef is not included in common.hpp and it is needed for size_t. I think we should keep it.

File: sycl/test-e2e/Matrix/element_wise_abc.cpp [hunk 2]
```diff
@@ -15,4 +15,5 @@
 #define SG_SZ 16
 constexpr size_t TN = 16;
 
+#include "common.hpp"
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/element_wise_abc.cpp:18]: In all other matrix tests we include common.hpp as one of the first header files, before we define global variables, so I suggest do the same for this file.

File: sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp (hunk 1 above)
Comment (artemrad) [reply sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp:14]: The tests pass. `size_t` is also included indirectly through most STL libraries. In this case it is included through ` #include <iostream>`, which is imported through `common.hpp`. See https://godbolt.org/z/Gh1GTefeW for an example.

File: sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp [hunk 3]
```diff
@@ -8,14 +8,10 @@
 
 // cuda backend test for joint_matrix_apply (and joint_matrix_fill)
 
-#include <sycl/sycl.hpp>
+#include "common.hpp"
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp:11]: If you are changing tests for CUDA, please, check that they are passing on CUDA after your change.

File: sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp [hunk 4]
```diff
@@ -6,9 +6,7 @@
 // Test checks that exception will be thrown in case object of joint_matrix type
 // is used on unsupported HW, in this case, on Gen12.
 
-#include <sycl/sycl.hpp>
-
-using namespace sycl::ext::oneapi::experimental::matrix;
+#include "common.hpp"
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp:9]: This test is running only on gen12, please, check that it is passing after your change.

File: sycl/test-e2e/Matrix/joint_matrix_query_default.cpp [hunk 5]
```diff
@@ -12,21 +12,7 @@
...
-template <typename T, size_t NUM_ROWS, size_t NUM_COLS> struct big_matrix {
-public:
-  T *mat;
-
-public:
-  T *get_data() { return mat; }
-  void set_data(T *data) { mat = data; }
-  big_matrix(T *data) : mat(data) {}
-};
+#include "common.hpp"
```
Comment (YuriPlyakhin) [inline sycl/test-e2e/Matrix/joint_matrix_query_default.cpp:15]: Also CPU specific tests, needs to be verified on CPU

**Review by YuriPlyakhin — CHANGES_REQUESTED** [review]
I noticed that tests in XMX8 folder were not changed, while some .hpp files which are used for XMX8 were changed. Most likely it will cause compilation failures for XMX8 tests. Please, update tests in that folder as well accordingly.

Also other comments need to be addressed.

File: sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp (hunk 1 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/SG32/element_wise_abc.cpp:14]: ok, then please, remove it from `sycl/test-e2e/Matrix/element_wise_abc.cpp` as well.

File: sycl/test-e2e/Matrix/joint_matrix_query_default.cpp (hunk 5 above)
Comment (artemrad) [reply sycl/test-e2e/Matrix/joint_matrix_query_default.cpp:15]: Passed see below.

File: sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp (hunk 4 above)
Comment (artemrad) [reply sycl/test-e2e/Matrix/joint_matrix_opt_kernel_feature_unsupported_hw.cpp:9]: Passed see below.

File: sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp (hunk 3 above)
Comment (artemrad) [reply sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp:11]: Passed see below.

File: sycl/test-e2e/Matrix/elemwise_irreg_size_ops_bf16.cpp [hunk 6]
```diff
@@ -14,12 +14,7 @@
 // RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
-#include <iostream>
-#include <sycl/sycl.hpp>
-
-using namespace sycl;
-using namespace sycl::ext::oneapi::experimental::matrix;
-using bfloat16 = sycl::ext::oneapi::bfloat16;
+#include "common.hpp"
```
[the remark this answers is not in the document]
Comment (artemrad) [reply sycl/test-e2e/Matrix/elemwise_irreg_size_ops_bf16.cpp:17]: Passed see below.

[the remark this answers is not in the document]
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/elemwise_irreg_size_ops_bf16.cpp:17]: I don't see this test was run at all

File: sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp (hunk 3 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_apply_cuda.hpp:11]: I don't see this test running at all

File: sycl/test-e2e/Matrix/joint_matrix_query_default.cpp (hunk 5 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_query_default.cpp:15]: I don't see this test was run at all.

**Review by YuriPlyakhin — CHANGES_REQUESTED** [review]
2 tests specific for CPU are not running in pre-check-in testing. Could you run them locally to verify your changes or remove these 2 tests from PR?

File: sycl/test-e2e/Matrix/elemwise_irreg_size_ops_bf16.cpp (hunk 6 above)
[the remark this answers is not in the document]
Comment (artemrad) [reply sycl/test-e2e/Matrix/elemwise_irreg_size_ops_bf16.cpp]: You are right  on both counts (I'm just figuring out what to look for):
* tests did not run
* my change broke the tests and it was not trivial to fix. Reverted the change.

File: sycl/test-e2e/Matrix/joint_matrix_query_default.cpp (hunk 5 above)
Comment (artemrad) [reply sycl/test-e2e/Matrix/joint_matrix_query_default.cpp]: See response above.

**Comment by dkhaldi** [comment]
Why did you remove the changes from the CPU tests?

**Review by YuriPlyakhin — APPROVED** [review]

**Comment by artemrad** [comment]
@dkhaldi  - I removed the changes to the CPU tests because it was not trivial to convert the CPU centric `matrix_multiply_ref()` to `common.hpp:matrix_multiply_ref()`. These tests can be updated and aligned with `common.hpp` with a future change

**Comment by artemrad** [comment]
@intel/llvm-gatekeepers Ready to merge

**Pull request merged**